app/src/main/java/com/example/moviedb/ui/nowplaying/NowPlayingFragment.kt

```diff
@@ -47,24 +45,22 @@ class NowPlayingFragment : BaseFragment<FragmentNowplayingBinding, NowPlayingVie
 
         if (recycler_movie.layoutManager is LinearLayoutManager) {
             val layoutManager = recycler_movie.layoutManager as LinearLayoutManager
-            recycler_movie.addOnScrollListener(object : EndlessScrollListener(layoutManager) {
-                override fun loadMore() {
-                    if (viewModel.loading.value == false) {
-                        viewModel.getMovies(page)
-                    }
+            recycler_movie.addOnScrollListener(EndlessScrollListener(layoutManager) {
+                if (viewModel.loading.value == false && viewModel.refreshData.value == false) {
+                    viewModel.getMovies(viewModel.currentPage)
                 }
             })
         }
 
         viewModel.getData().observe(viewLifecycleOwner, Observer { movies ->
             adapter.submitList(movies)
         })
-        viewModel.isSuccess.observe(viewLifecycleOwner, Observer { success ->
-            if (success) {
-                page++
+        viewModel.refreshData.observe(viewLifecycleOwner, Observer { refresh ->
+            if (refresh != null) {
```

Comment:
Có thể viết thế này: refresh?.let{                swipe_refresh_layout.isRefreshing = it
}